io/include/pcl/compression/impl/octree_pointcloud_compression.hpp

```diff
@@ -359,6 +359,12 @@ namespace pcl
           // decode differential color information
           std::vector<char>& pointDiffColorDataVector = color_coder_.getDifferentialDataVector ();
           compressed_tree_data_in_arg.read (reinterpret_cast<char*> (&point_diff_color_data_vector_size), sizeof (point_diff_color_data_vector_size));
+          PCL_CHECK_IO_STREAM(compressed_tree_data_in_arg, "size of pointDiffColorDataVector");
+
+          if (point_diff_color_data_vector_size < 0)
+          {
+            PCL_THROW_EXCEPTION (pcl::IOException, "Error! Size of pointDiffColorDataVector specified in the file is negative!");
```

Comment:
This exception is complaining about a bad/incorrect numerical value. It's not really an IO issue anymore. This is the wrong type of exception to throw in this situation, but unfortunately I don't see an existing PCL Exception which matches what we would likely want to express here. Maybe `BadArgumentException`? Otherwise I'm open to create a new exception type and I'm looking forsuggestions on a name which better expresses what is happening here.

Reply:
I don't think `BadArgumentException` is suited for this situation. Perhaps a new exception `BadValueInFileException`?

Reply:
Maybe just `BadValue` so that it can be used in other contexts.